### DISCUSSION TITLE
Starting point for Thread 1.3 Border Router running on FreeRTOS

### DISCUSSION THREAD

**Discussion by hyukdong** [openthread/openthread#8917] [Q&A] [answered]
Hello, 
I need help on Thread 1.3 Border Router, summarized as follows.

Objectives: Build Thread 1.3.0-compliant Border Router (Host SW) running on FreeRTOS
Question: Where can i get the appropriate source code among the following 3 candidate github repositories?
  1) openthread (this repository)
    - till now I've been thinking ot-br-posix is the only repository that provides Border Router functionalities,
      but it seems that openthread can also provide Border Router functionalities. Am I right?
  2) ot-rtos
    - It would be nice if I can get Thread 13 BR from ot-rtos, as RTOS is my target platform.
       but not sure if it has border agent functions nor Thread 1.3 compliant FTD functions. 
  3) ot-br-posix
    - I' managed to build Thread 1.3.0 border router by giving -DOT_THREAD_VERSION=1.3 for OTBR_OPTIONS,
      but it does not seem to provide DNS-SD & SRP which is required for Thread 1.3.

Thank you for your consideration of this question.

**Comment by wgtdkp** [comment, marked as answer]
1. openthread
    - Many of the 1.3 Border Router features are implemented in openthread, such as the [SRP server](https://github.com/openthread/openthread/blob/main/src/core/net/srp_server.cpp) and [Routing Manager](https://github.com/openthread/openthread/blob/main/src/core/border_router/routing_manager.cpp).
2. ot-rtos
    - ot-rtos includes early (prior to 1.3) basic Thread Border Router features such as IP forwarding, there isn't any 1.3 related feature implemented in the ot-rtos repo.
3. ot-br-posix
     - it implements the platform APIs which are needed for building a 1.3 BR on posix platforms.
     - it implements the Advertising Proxy and Discovery Proxy

To try 1.3 Border Router with ot-br-posix, please follow the codelab: https://openthread.io/codelabs/openthread-border-router

To implement a 1.3 Border Router on freeRTOS, you will minimumly need to
1. Implement the RTOS version of the platform APIs which has been implemented for posix in [src/posix/platform](https://github.com/openthread/openthread/tree/main/src/posix/platform)
2. Implement the Discovery and Advertising Proxy which has been implemented in [ot-br-posix/src/sdp_proxy](https://github.com/openthread/ot-br-posix/tree/main/src/sdp_proxy)

FYI, Espressif open sourced their Thread Border Router solution https://github.com/espressif/esp-thread-br but unfortunately, the core Border Router feature implementations are released as pre-built binaries.

**Reply by hyukdong** [reply]
Thank you for the detailed reply. It really helped me out
